### ISSUE TITLE
Wildcard support for subfields seems ignoring undefined fields

### ISSUE THREAD

**Issue by xuefengwang** [ctavan/express-validator#458]
Here is my test case for this scenario. The code is supposed to trigger missing addresses[1].address1, but it actually passed validation. I'm using express-validator 4.2.1.

Route handler:
```
router.post('/', [
    check('addresses.*.address1').exists(),
    check('addresses.*.city').exists()
], (req, res, next) => {
    const errors = validationResult(req);
    if (!errors.isEmpty()) {
        console.log(errors.mapped());
        res.status(400).json({errors: errors.mapped()});
    } else {
        res.status(200).json({title: 'Express'});
    }
});
```

Test case:
```
describe('index', function () {
    it('should throw invalid address', (done) => {
        chai.request('http://localhost:3000')
            .post('/')
            .send({
                'addresses': [ { 'address1': '123 St', 'city': 'Test' }, { 'city': 'Test' } ]
            })
            .end((err, res) => {
                res.should.have.status(400);
                res.body.should.be.a('object');
                res.body.should.have.property('errors');
                res.body.errors.should.have.property('addresses[1].address1');
                res.body.errors['addresses[1].address1'].msg.should.equal('Invalid value');
                done();
            });
    });
});
```

Thanks,
William

**Comment by xuefengwang** [comment]
After some debugging into the code, it seems in select-fields.js, as below
```
    if (instances.length > 1) {
      const withValue = instances.filter(field => field.value !== undefined);
      instances = withValue.length ? withValue : [instances[0]];
    }
```
addresses[1].address1 was in the instances. But after the logic above, since the value is undefined for addresses[1].address1, then it got filtered out.

**Comment by gustavohenke** [comment]
Thanks for finding this out!
This if statement is there for cases when you have the same field in more than one location. In such cases, they work fine, but this also affects legitimate cases like yours.

It seems like a complex bug 😕

**Comment by xuefengwang** [comment]
Thanks for looking into this, @gustavohenke. As you mentioned, the code is trying to get undefined from other locations. For the timing being, as a hack/workaround, I changed code locally as below to make it work. 
```
    if (instances.length > 1 && field.indexOf('*') === -1) {
      const withValue = instances.filter(field => field.value !== undefined);
      instances = withValue.length ? withValue : [instances[0]];
    }
```

**Comment by gustavohenke** [comment]
Partial fix merged in #492.
Now if you use location-specific functions (eg `body`, `params`, etc instead of `check`), the bug will not happen. Room for improvement still exists though...

**Comment by gustavohenke** [comment]
Published [`v5.2.0`](https://github.com/express-validator/express-validator/releases/tag/v5.2.0) 🚢
